[GLib] Remove all GLIB_CHECK_VERSION conditionals

aperezdc opened the pull request:
#### 85b637b69f1c3a6242420b198d1c173477ce0f22
<pre>
[GLib] Remove all GLIB_CHECK_VERSION conditionals
<a href="https://bugs.webkit.org/show_bug.cgi?id=286061">https://bugs.webkit.org/show_bug.cgi?id=286061</a>

Reviewed by Miguel Gomez.

Among the Linux distributions that will be supported in 2025 the oldest
version of GLib in use is 2.72 (packaged in Ubuntu 22.04). This means it
is now possible to remove all the existing GLib version checks and
increase the minimum GLib version required by CMake to 2.70 instead.

While at it, remove the HAVE(GURI) conditionals: GUri was introduced in
GLib 2.66, and has had all the functionality required for WebKit usage
since version 2.68.

* Source/WTF/wtf/URL.h:
* Source/WTF/wtf/glib/GRefPtr.cpp:
(WTF::derefGPtr):
* Source/WTF/wtf/glib/GRefPtr.h:
* Source/WTF/wtf/glib/SocketConnection.cpp:
(WTF::SocketConnection::readMessage):
* Source/WTF/wtf/glib/URLGLib.cpp:
(WTF::URL::createGUri const):
* Source/WebCore/platform/LowPowerModeNotifier.h:
* Source/WebCore/platform/glib/LowPowerModeNotifierGLib.cpp:
(WebCore::LowPowerModeNotifier::LowPowerModeNotifier):
(WebCore::LowPowerModeNotifier::~LowPowerModeNotifier):
* Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp:
(WebCore::CertificateInfo::isolatedCopy const):
(WebCore::CertificateInfo::summary const):
* Source/WebKit/NetworkProcess/glib/DNSCache.cpp:
(WebKit::DNSCache::mapForType):
(WebKit::DNSCache::removeExpiredResponsesFired):
(WebKit::DNSCache::clear):
* Source/WebKit/NetworkProcess/glib/DNSCache.h:
* Source/WebKit/NetworkProcess/glib/WebKitCachedResolver.cpp:
(webkitCachedResolverLookupByNameWithFlagsFinish):
(webkit_cached_resolver_class_init):
* Source/WebKit/NetworkProcess/glib/WebKitOverridingResolver.cpp:
(webkitOverridingResolverLookupByNameWithFlagsFinish):
(webkit_overriding_resolver_class_init):
* Source/WebKit/Shared/glib/ArgumentCodersGLib.cpp:
(IPC::ArgumentCoder&lt;GRefPtr&lt;GTlsCertificate&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;GRefPtr&lt;GTlsCertificate&gt;&gt;::decode):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/289318@main">https://commits.webkit.org/289318@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f8d2df3324fe9ee4df8a4b933922b9ee86ac6412

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86373 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5925 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/40710 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91286 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37176 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/88428 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6179 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13962 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66884 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24675 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89380 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4693 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78237 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47205 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4503 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/32554 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36293 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/79223 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75039 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33426 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93162 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/85211 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/13577 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9854 "Found 1 new test failure: fast/mediastream/get-user-media-on-loadedmetadata.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75694 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/13785 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74108 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74876 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19116 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17507 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/6394 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13445 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/13601 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/18888 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/107677 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13354 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25915 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/16797 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15138 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dpino commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -214,7 +214,7 @@ endif ()
     set(GLIB_MINIMUM_VERSION 2.70.0)
 else ()
-    set(GLIB_MINIMUM_VERSION 2.56.4)
+    set(GLIB_MINIMUM_VERSION 2.69.1)
```
Since Ubuntu 22.04 features GLib 2.72.1, and it's currently the oldest OS we need to support, why not requiring that version of GLib regardless of whether `ENABLE_2022_GLIB_API` is ON or OFF?

aperezdc replied:
I think it's good to keep the requirement as low as our code needs, even when we know that the oldest supported distribution has a newer version. Let's not artificially have a higher requirement, we never know when somebody might need to try building on a system that has GLib 2.70 😉 

What I am going to do is change the dependency to always be 2.70.0, because 2.69.1 is a development snapshot, as @carlosgcampos pointed out elsewhere.

aperezdc replied:
Done!

aperezdc committed:
[GLib] Remove all GLIB_CHECK_VERSION conditionals
https://bugs.webkit.org/show_bug.cgi?id=286061

Reviewed by Miguel Gomez.

Among the Linux distributions that will be supported in 2025 the oldest
version of GLib in use is 2.72 (packaged in Ubuntu 22.04). This means it
is now possible to remove all the existing GLib version checks and
increase the minimum GLib version required by CMake to 2.70 instead.

While at it, remove the HAVE(GURI) conditionals: GUri was introduced in
GLib 2.66, and has had all the functionality required for WebKit usage
since version 2.68.

* Source/WTF/wtf/URL.h:
* Source/WTF/wtf/glib/GRefPtr.cpp:
(WTF::derefGPtr):
* Source/WTF/wtf/glib/GRefPtr.h:
* Source/WTF/wtf/glib/SocketConnection.cpp:
(WTF::SocketConnection::readMessage):
* Source/WTF/wtf/glib/URLGLib.cpp:
(WTF::URL::createGUri const):
* Source/WebCore/platform/LowPowerModeNotifier.h:
* Source/WebCore/platform/glib/LowPowerModeNotifierGLib.cpp:
(WebCore::LowPowerModeNotifier::LowPowerModeNotifier):
(WebCore::LowPowerModeNotifier::~LowPowerModeNotifier):
* Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp:
(WebCore::CertificateInfo::isolatedCopy const):
(WebCore::CertificateInfo::summary const):
* Source/WebKit/NetworkProcess/glib/DNSCache.cpp:
(WebKit::DNSCache::mapForType):
(WebKit::DNSCache::removeExpiredResponsesFired):
(WebKit::DNSCache::clear):
* Source/WebKit/NetworkProcess/glib/DNSCache.h:
* Source/WebKit/NetworkProcess/glib/WebKitCachedResolver.cpp:
(webkitCachedResolverLookupByNameWithFlagsFinish):
(webkit_cached_resolver_class_init):
* Source/WebKit/NetworkProcess/glib/WebKitOverridingResolver.cpp:
(webkitOverridingResolverLookupByNameWithFlagsFinish):
(webkit_overriding_resolver_class_init):
* Source/WebKit/Shared/glib/ArgumentCodersGLib.cpp:
(IPC::ArgumentCoder<GRefPtr<GTlsCertificate>>::encode):
(IPC::ArgumentCoder<GRefPtr<GTlsCertificate>>::decode):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/289318@main
